---
title: "Critical evaluation of the reliability of DNA methylation probes on the Illumina MethylationEPIC BeadChip microarrays for dementia research"
subtitle: "Compare ICC in two groups"
author: "Wei Zhang, Lily Wang"
date: "`r Sys.Date()`"
output:
  html_document:
    highlight: pygments
    theme: yeti
    toc: true
    number_sections: true
    df_print: paged
    code_download: false
    toc_float:
      collapsed: yes
    toc_depth: 3
    font_size: 10
editor_options:
  chunk_output_type: inline 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(coMethDMR)
library(geepack)
library(lme4)
library(lmerTest)
library(fgsea)
library(emmeans)
library(tidyverse)

dir.base <- "../../"
dir.reliable.score <- file.path(dir.base, "analysis-results/reliable_score_different_plate")
dir.data.auc <- file.path(dir.base, "data/Aux_Sync")
dir.data.supp <- file.path(dir.base, "data/SuppFiles")
dir.results.test <- file.path(dir.base, "analysis-results/compare_test")
dir.RNA_vs_DNAm <- file.path(dir.base, "analysis-results/RNA_vs_DNAm")

dir.mike <- file.path(dir.base,  "Mike")
dir.extra <- file.path(dir.base, "external_files")
for(p in grep("dir",ls(),value = T)) dir.create(get(p),recursive = TRUE,showWarnings = FALSE)
```


# Load results

```{r message = F}
# Load beta values of first replicants
beta.rep1 <- read_csv(
  file.path(dir.data.auc, "ADNI_beta_duplicated_samples_diff_plates_replication_1.csv")
) %>% column_to_rownames("CpG")
colnames(beta.rep1) <- gsub("X", "", colnames(beta.rep1))
# Load reliable results
reliable_df <- read_csv(file.path(dir.reliable.score, "ADNI_ICC_annotation_with_HoLA_grouped.csv"))
```


# Identify co-methylated clusters using first replicate

```{r eval = F}
# Pre-defined region
probes.cluster.all <- CloseBySingleRegion(
  rownames(beta.rep1),
  genome = "hg19",
  arrayType = "EPIC"
)
# Find co-methylated clusters
coMeth_ls <- CoMethAllRegions(  
  dnam = beta.rep1,
  CpGs_ls = probes.cluster.all,
  arrayType = "EPIC",
  method = "spearman",
  returnAllCpGs = TRUE,              
  output = "CpGs",
  nCores_int = 5,
  progressbar = TRUE
)
saveRDS(
  coMeth_ls,
  file.path(dir.data.auc, "coMethDMR_region_ADNI_beta_replication_1.RDS")
)
```

```{r eval = F}
# combine with reliable results
reliable_df_with_region <- plyr::ldply(
  1:length(coMeth_ls),
  .fun = function(ls){
    
    cpg_in_region <- coMeth_ls[[ls]]
    reliable_df %>% 
      filter(cpg %in% cpg_in_region) %>% 
      mutate(cluster = ls, region = names(coMeth_ls)[ls]) %>% 
      dplyr::select(cpg, cluster, region)
    
    
  }
)
reliable_df_with_region2 <- left_join(reliable_df, reliable_df_with_region)
reliable_df_with_region2$cluster[is.na(reliable_df_with_region2$cluster)] <- 0
# ------------------------------------------------------------------------------------------------------------
# Save results
# ------------------------------------------------------------------------------------------------------------
write_csv(
  reliable_df_with_region2,
  file.path(dir.reliable.score, "ADNI_ICC_annotation_with_HoLA_grouped_and_region_cluster.csv")
)
```

# Compare test analysis

```{r message = F}
reliable_df_with_region2 <- read_csv(
  file.path(dir.reliable.score, "ADNI_ICC_annotation_with_HoLA_grouped_and_region_cluster.csv")
)
reliable_df_with_region2 <- reliable_df_with_region2 %>% separate(col = "UCSC_RefGene_Name", into = "Gene", sep = ";", remove = F)
reliable_df_with_region2$Gene[is.na(reliable_df_with_region2$Gene)] <- "none"
```

```{r}
# ------------------------------------------------------------------------------------------------------------
# Anxilary results
# ------------------------------------------------------------------------------------------------------------
model.fn <- function(data, Type = "ICC_Beta", test_var = NULL, model = "GEE", method = "del.eff"){
  
  data[[test_var]] <- factor(data[[test_var]])
  
  if(model == "GEE"){
    
    formu <- formula(paste(Type, "~ -1 + ", test_var))
    results <- geeglm(
      formu, 
      data = data,
      id = cluster,
      corstr = "exchangeable"
    )
    
  }
  
  if(model == "LMER"){
    
    formu <- formula(paste(Type, "~ -1 +", test_var, "+ (1 | cluster) + (1 | Gene) + (1 | seqnames)"))
    results <- lmer(
      formu, 
      data = data
    )

    
  }
  
    e <- emmeans(results, specs = test_var)
    c_df <- data.frame(Type = Type, contrast(e, type = "average", method = method))   # method = "del.eff"
    c_df$contrast <- gsub(" effect","",c_df$contrast)
    c_df$df <- nrow(data)
    c_df$p.t <- 2 * (1 - pt(abs(c_df$z.ratio), c_df$df))
  
  return(c_df)
  
}
```

## ICC

```{r eval = F}
test_v <- c("Relation_to_Island", "RefGene_Group", "Probe_Type")
ICC_test_results <- plyr::llply(
  test_v,
  .fun = function(t){
    
    model.fn(data = reliable_df_with_region2, 
             Type = "ICC_Beta", 
             test_var = t, 
             model = "LMER")
    
  }
)
names(ICC_test_results) <- test_v
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "ICC_vs_groups_LMER_model_results.xlsx")
)
```


## modified ICC

```{r eval = F}
# Test on mean group and sd group
q0 <- quantile(reliable_df_with_region2$beta.sd.rand)
q <- paste0("Q" , 1:4, ": ", 
            sapply(1:4, function(k) paste0(round(q0[k],2), "-", round(q0[k + 1],2))))
reliable_df_with_region2 <- reliable_df_with_region2 %>% 
  mutate(group.mean = ifelse(beta.mean < 0.2, "0 - 0.2",
                              ifelse(beta.mean < 0.4 & beta.mean >= 0.2, "0.2 - 0.4",
                                     ifelse(beta.mean < 0.6 & beta.mean >= 0.4, "0.4 - 0.6", 
                                            ifelse(beta.mean < 0.8 & beta.mean >= 0.6, "0.6 - 0.8", "0.8 - 1")))),
         group.mean = factor(group.mean, 
                              levels = c("0 - 0.2", 
                                         "0.2 - 0.4",
                                         "0.4 - 0.6",
                                         "0.6 - 0.8", 
                                         "0.8 - 1")),
         Quantile.sd = ifelse(beta.sd.rand < q0[2], q[1],
                              ifelse(beta.sd.rand < q0[3] & beta.sd.rand >= q0[2], q[2],
                                     ifelse(beta.sd.rand < q0[4] & beta.sd.rand >= q0[3], q[3], q[4]))),
         Quantile.sd = factor(Quantile.sd, levels = q))

test_v <- c("group.mean", "Quantile.sd", "Relation_to_Island", "RefGene_Group", "Probe_Type")
ICC_test_results <- plyr::llply(
  test_v,
  .fun = function(t){
    
    model.fn(data = reliable_df_with_region2, 
             Type = "modified_ICC", 
             test_var = t, 
             model = "LMER")
    
  }
)
names(ICC_test_results) <- test_v

writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_groups_LMER_model_results.xlsx")
)
```

### mICC vs mQTLs

```{r}
ICC_test_results <- model.fn(
  data = reliable_df_with_region2,
  Type = "modified_ICC",
  test_var = "mqtl",
  model = "LMER"
)
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_mQTLs_LMER_model_results.xlsx")
)
```

### mICC vs ADNI studies

```{r}
ICC_test_results <- list()
```


```{r message = F}
reliable_df <- read_csv(
  file.path(dir.reliable.score, "ADNI_ICC_beta_and_M_results_ind_diff_plates_samples_annotated_with_ADNI_AIBL.csv")
)
reliable_df <- left_join(reliable_df, (reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene)))
```

```{r}
ICC_test_results[[1]] <- model.fn(
  data = reliable_df,
  Type = "effect_size_diff",
  test_var = "m_ICCgroup",
  model = "LMER"
)
```

```{r message = F}
reliable_df <- read_csv(
  file.path(dir.reliable.score, "FEMALE_ICC_beta_and_M_results_ind_diff_plates_samples_annotated_with_ADNI_AIBL.csv")
)
reliable_df <- left_join(reliable_df, (reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene)))
```

```{r}
ICC_test_results[[2]] <- model.fn(
  data = reliable_df,
  Type = "effect_size_diff",
  test_var = "m_ICCgroup",
  model = "LMER"
)
```

```{r message = F}
reliable_df <- read_csv(
  file.path(dir.reliable.score, "MALE_ICC_beta_and_M_results_ind_diff_plates_samples_annotated_with_ADNI_AIBL.csv")
)
reliable_df <- left_join(reliable_df, (reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene)))
```

```{r}
ICC_test_results[[3]] <- model.fn(
  data = reliable_df,
  Type = "effect_size_diff",
  test_var = "m_ICCgroup",
  model = "LMER"
)
```

```{r}
names(ICC_test_results) <- c("mICC_vs_combined_sex", "mICC_vs_female", "mICC_vs_male")
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_ADNI_study_LMER_model_results.xlsx")
)
```

### mICC vs brain blood correlation


```{r}
reliable_df <- read_csv( file.path(dir.reliable.score,"ADNI_ICC_beta_and_M_results_ind_diff_plates_samples_annotated_with_ADNI_AIBL_brain_blood_corr.csv"))
reliable_df <- left_join(reliable_df, (reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene)))
reliable_df$beta_spearman_cor_group <- factor(reliable_df$beta_spearman_cor_group, levels = c("Low", "Medium", "High"))
```

```{r}
ICC_test_results <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "beta_spearman_cor_group",
  model = "LMER"
)
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_brain_blood_correlation_LMER_model_results.xlsx")
)
```

### mICC vs DMR

```{r}
ICC_test_results <- list()
```


```{r message = F}
DMR <- readxl::read_xlsx(file.path(dir.mike, "cpg_plus3.xlsx"))
reliable_df <- left_join(DMR, (reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene)))
reliable_df$located_in_combined <- ifelse(reliable_df$located_in_combined == 1, "Yes", "No")
```

```{r}
ICC_test_results[[1]] <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "located_in_combined",
  model = "LMER"
)
```

```{r}
reliable_df$located_in_females <- ifelse(reliable_df$located_in_females == 1, "Yes", "No")
```

```{r}
ICC_test_results[[2]] <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "located_in_females",
  model = "LMER"
)
```

```{r}
reliable_df$located_in_males <- ifelse(reliable_df$located_in_males == 1, "Yes", "No")
```

```{r}
ICC_test_results[[3]] <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "located_in_males",
  model = "LMER"
)
```

```{r}
names(ICC_test_results) <- c("Combined", "Female", "Male")
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_DMR_LMER_model_results.xlsx")
)
```


### mICC vs DNAm RNA correlation

```{r}
ICC_test_results <- list()
```


```{r}
# promoter
reliable_df <- read_csv(file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_promoter_region.csv"))
reliable_df <- left_join(reliable_df, reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene))
reliable_df <- reliable_df %>% 
  mutate(sig = ifelse(p.val < 1e-05, "Yes", "No"))
```

```{r}
ICC_test_results[[1]] <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "sig",
  model = "LMER"
)
```

```{r}
# distal
reliable_df <- read_csv(file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_distal_region.csv"))
reliable_df <- left_join(reliable_df, reliable_df_with_region2 %>% dplyr::select(cpg, cluster, Gene))
reliable_df <- reliable_df %>% 
  mutate(sig = ifelse(p.val < 1e-05, "Yes", "No"))
```

```{r}
ICC_test_results[[2]] <- model.fn(
  data = reliable_df,
  Type = "modified_ICC",
  test_var = "sig",
  model = "LMER"
)
```

```{r}
names(ICC_test_results) <- c("mICC_vs_promoter", "mICC_vs_distal")
```

```{r}
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "modified_ICC_vs_DNAm_RNA_correlation_LMER_model_results.xlsx")
)
```

## ICC vs other study

```{r}
# ------------------------------------------------------------------------------------------------------------
# Anxilary results
# ------------------------------------------------------------------------------------------------------------
model.fn <- function(data, Type = "ICC_Beta", test_var = NULL, model = "LMER", method = "del.eff"){
  
  data[[test_var]] <- factor(data[[test_var]])
  
  if(model == "GEE"){
    
    formu <- formula(paste(Type, "~ -1 + ", test_var))
    results <- geeglm(
      formu, 
      data = data,
      id = cluster,
      corstr = "exchangeable"
    )
    
  }
  
  if(model == "LMER"){
    
    formu <- formula(paste(Type, "~ -1 +", test_var, "+ (1 | cluster) + (1 | Gene) + (1 | seqnames) + (1 | cpg)"))
    results <- lmer(
      formu, 
      data = data
    )

    
  }
  
    e <- emmeans(results, specs = test_var)
    c_df <- data.frame(Type = Type, contrast(e, type = "average", method = method))   # method = "del.eff"
    c_df$contrast <- gsub(" effect","",c_df$contrast)
    c_df$df <- nrow(data)
    c_df$p.t <- 2 * (1 - pt(abs(c_df$z.ratio), c_df$df))
  
  return(c_df)
  
}
```

### Sugden

```{r}
reliable_CpG_Sugden <- readxl::read_xlsx(
  file.path(dir.data.supp, "1-s2.0-S2666389920300143-mmc2.xlsx"),
  skip = 2
) 
colnames(reliable_CpG_Sugden)[1] <- "cpg"
reliable_CpG_Sugden <- reliable_CpG_Sugden %>% dplyr::mutate(Sugden_classification = factor(ifelse(Reliability < 0.4, "Poor", 
                                            ifelse(Reliability >= 0.4 & Reliability < 0.6, "Fair",
                                                   ifelse(Reliability >= 0.6 & Reliability < 0.75, "Good", "Excellent"))),
                                            levels = c("Poor", "Fair", "Good", "Excellent"))) 
reliable_score.S <- inner_join(reliable_df_with_region2, reliable_CpG_Sugden) %>% 
  dplyr::select(c("cpg", "ICC_Beta", "Reliability", "cluster", "Gene", "seqnames")) %>%
  pivot_longer(
    cols = c("ICC_Beta", "Reliability"),
    names_to = "Study",
    values_to = "ICC"
) %>% mutate(Study = ifelse(Study == "ICC_Beta", "ADNI", "Sugden"))
```

```{r}
ICC_test_results <- list()
```

```{r}
ICC_test_results[[1]] <- model.fn(
  reliable_score.S,
  Type = "ICC",
  test_var = "Study",
  model = "LMER"
)
```

### Bose

```{r}
reliable_CpG_Bose <- read_csv(
  file.path(dir.data.supp, "Bose-ARIC-ICC-values.csv")
) 
colnames(reliable_CpG_Bose)[1] <- "cpg"
reliable_CpG_Bose <- reliable_CpG_Bose %>% dplyr::select(cpg, `ICC value`) %>%
  dplyr::mutate(Bose_classification = factor(ifelse(`ICC value` < 0.4, "Poor", 
                                            ifelse(`ICC value` >= 0.4 & `ICC value` < 0.6, "Fair",
                                                   ifelse(`ICC value` >= 0.6 & `ICC value` < 0.75, "Good", "Excellent"))),
                                            levels = c("Poor", "Fair", "Good", "Excellent"))) 

reliable_score.B <- inner_join(reliable_df_with_region2, reliable_CpG_Bose) %>% 
  dplyr::select(c("cpg", "ICC_Beta", "ICC value", "cluster", "Gene", "seqnames")) %>%
  pivot_longer(
    cols = c("ICC_Beta", "ICC value"),
    names_to = "Study",
    values_to = "ICC"
) %>% mutate(Study = ifelse(Study == "ICC_Beta", "ADNI", "Bose"))
```


```{r}
ICC_test_results[[2]] <- model.fn(
  reliable_score.B,
  Type = "ICC",
  test_var = "Study",
  model = "LMER"
)
names(ICC_test_results) <- c("ICC_vs_Sugden", "ICC_vs_Bose")
writexl::write_xlsx(
  ICC_test_results,
  file.path(dir.results.test, "ICC_vs_other_study_LMER_model_results.xlsx")
)
```

