---
title: "Critical evaluation of the reliability of DNA methylation probes on the Illumina MethylationEPIC BeadChip microarrays for dementia research"
subtitle: "Correlation with gene expression"
author: "Wei Zhang, Lily Wang"
date: "`r Sys.Date()`"
output:
  html_document:
    highlight: pygments
    theme: yeti
    toc: true
    number_sections: true
    df_print: paged
    code_download: false
    toc_float:
      collapsed: yes
    toc_depth: 3
    font_size: 10
editor_options:
  chunk_output_type: inline 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(data.table)
library(matrixStats)
library(SummarizedExperiment)
library(MethReg)
library(dorothea)
library(readr)
library(ggpubr)
library(hgu219.db)

dir.base <- "../../"
dir.data.new <- file.path(dir.base, "data/ADNI")
dir.data.new.processed <- file.path(dir.data.new, "DNA_methylation/processed")
dir.reliable.score <- file.path(dir.base, "analysis-results/reliable_score_different_plate")
dir.plot <- file.path(dir.base, "analysis-results/plots")
dir.RNA_vs_DNAm <- file.path(dir.base,"analysis-results/RNA_vs_DNAm")
dir.data.auc <- file.path(dir.base,"data/Aux_Sync") 

dir.extra <- file.path(dir.base, "external_files")
for(p in grep("dir",ls(),value = T)) dir.create(get(p),recursive = TRUE,showWarnings = FALSE)
```


# Load results

```{r message = F, eval = F}
# Load reliable results
reliable_df <- read_csv(file.path(dir.reliable.score, "ADNI_icc_beta_and_M_results_ind_samples_annotated_different_plates.csv"))
```

# Matched RNA DNAm data

```{r message=F, eval = F}
# -----------------------------------------------------------------------------
# Gene expression: Affymetrix Human Genome U 219 array 
# -----------------------------------------------------------------------------
ADNI_Gene_Expression_Profile <- read_csv(
  file.path(dir.extra, "ADNI_Gene_Expression_Profile.csv"),
    skip = 8
  )
  colnames(ADNI_Gene_Expression_Profile) <- gsub("\\...", "X", colnames(ADNI_Gene_Expression_Profile))
  ADNI_Gene_Expression_Profile$X748 <- NULL
  
ADNI_Gene_Expression_Profile <- ADNI_Gene_Expression_Profile %>% tidyr::separate_rows("Symbol")

x <- hgu219ENSEMBL
# Get the entrez gene IDs that are mapped to an Ensembl ID, this is required by MethReg
probe.to.ensg <- as.data.frame(as.list(x) %>% unlist) %>% na.omit
ADNI_Gene_Expression_Profile$ENGS <- probe.to.ensg[ADNI_Gene_Expression_Profile$ProbeSet,]
ADNI_Gene_Expression_Profile <- ADNI_Gene_Expression_Profile[!is.na(ADNI_Gene_Expression_Profile$ENGS),]

nrow(ADNI_Gene_Expression_Profile) # 43890

# dropping genes in bottom 10 percentile for over 80% of the samples
genes.low.expressed.samples.count <- plyr::aaply(
  ADNI_Gene_Expression_Profile[,grep("^X",colnames(ADNI_Gene_Expression_Profile))] ,
  .margins = 2, # for each sample set get all genes
  .fun = function(sample.genes){
    # for each sample, mark the bottom 10% less expressed genes as 1
    sample.genes[[1]]  <= quantile(sample.genes[[1]] , probs = c(.10),type = 3)
  }) %>% colSums()
genes.idx <- which(genes.low.expressed.samples.count > (length(grep("^X",colnames(ADNI_Gene_Expression_Profile))) * 0.8))
ADNI_Gene_Expression_Profile <- ADNI_Gene_Expression_Profile[-c(genes.idx),]

# Since we have multiple probes mapping to the same gene in the array 
# we will take the median of the same genes
# as suggest in https://www.nature.com/articles/s41598-020-60595-1
expression.matrix <- plyr::aaply(unique(ADNI_Gene_Expression_Profile$ENGS),.margins = 1,.fun = function(gene){
  dat <- ADNI_Gene_Expression_Profile[ADNI_Gene_Expression_Profile$ENGS == gene,grep("^X",colnames(ADNI_Gene_Expression_Profile))]
  colMedians(dat %>% as.matrix)
},.progress = "time")
rownames(expression.matrix) <- unique(ADNI_Gene_Expression_Profile$ENGS)

ADNI_Gene_Expression_Metadata <- read_csv(
  file.path(dir.extra, "ADNI_Gene_Expression_Profile.csv"),
  skip = 0,col_names = FALSE,n_max = 7
)
ADNI_Gene_Expression_Metadata$X748 <- NULL

gene.exp.IDs <- apply(
  ADNI_Gene_Expression_Metadata[,4:ncol(ADNI_Gene_Expression_Metadata)],MARGIN = 2,
  FUN = function(col) {
    paste0(
      stringr::str_extract(pattern = "[0-9]*$",string = col[3]),
      "_", col[1],"_",
      col[2]
    )
  }
)

DXSUM <- readr::read_csv(
  file.path(dir.extra, "DXSUM_PDXCONV_ADNIALL_downloaded_2-19-2021.csv")
)
DXSUM <- DXSUM[match(gene.exp.IDs,paste0(stringr::str_extract(pattern = "[0-9]*$",DXSUM$PTID),"_",DXSUM$Phase,"_",DXSUM$VISCODE)),]
gene.exp.IDs <- paste0(DXSUM$RID,"_",DXSUM$Phase,"_",DXSUM$VISCODE2)
colnames(expression.matrix) <- gene.exp.IDs
expression.matrix <- expression.matrix[,colnames(expression.matrix) != "NA_NA_NA"]
```

```{r message = F, eval = F}
#-----------------------------------------------------------------------------
# get matched DNAm and Gene expression
#-----------------------------------------------------------------------------
beta <- readRDS( file.path(dir.data.new.processed,"ADNI_RnBeads_and_pooBAH_filtered_beta_different_plates.RDS"))
clinical <- read_csv(file.path(dir.data.new.processed, "sample_annotation.csv"))
### match with the current samples
clinical <- clinical %>% filter(barcodes %in% colnames(beta))
clinical <- clinical[match(colnames(beta), clinical$barcodes),]
dnam.IDs <- paste0(clinical$RID,"_",clinical$COLPROT,"_",clinical$VISCODE)
names(dnam.IDs) <- clinical$barcodes
table(gene.exp.IDs %in% dnam.IDs)
# FALSE  TRUE
#  713    31 
common.ids <- base::intersect(dnam.IDs %>% as.character,gene.exp.IDs %>% as.character)
barcodes <- dnam.IDs[match(common.ids,dnam.IDs)] %>% names
beta.select <- beta[,match(barcodes, colnames(beta))]
expression.matrix <- expression.matrix[,colnames(expression.matrix) %in% common.ids]
expression.matrix <- expression.matrix[,match(common.ids,colnames(expression.matrix))]

save(
  beta.select,
  expression.matrix,
  ADNI_Gene_Expression_Metadata,
  file = file.path(dir.data.auc,"ADNI_replicants_samples_diff_plates_matched_rna_dnam.rda")
)
```

# DNAm to RNA association

```{r}
#-------------------------------------------------------------------------------
# Aux functions
#-------------------------------------------------------------------------------
auxfunction <- function(
  row,
  exp = NULL,
  dnam = NULL
){
  
  rna.target <- exp[which(rownames(exp) == row$target), , drop = FALSE]
  met <- dnam[which(rownames(dnam) == as.character(row$regionID)), , drop = FALSE]
  
  # Calulate spearman correlations and p value using cor.test
  results <- cor.test(rna.target %>% as.numeric() , met %>% as.numeric(), method = "spearman", exact = F)
  results.df <- data.frame(rho = results$estimate, p.val = results$p.value)
  
  return(results.df)
}

add_cpgs_from_and_do_fdr <- function(results){
  
  results$fdr <- p.adjust(results$p.val, method = "fdr")
  
  return(results)
}
```


```{r eval = F}
#-------------------------------------------------------------------------------
# Map probes to regions
#-------------------------------------------------------------------------------
dnam <- MethReg:::map_probes_to_regions(
  dnam = beta.select[grep("^cg",rownames(beta.select)),],
  genome = "hg19",
  arrayType = "EPIC",
  rm.masked.probes = FALSE
)
```


## Promoter analysis

### Calculate correlations

```{r eval = F}
#-------------------------------------------------------------------------------
# Promoter analysis, triplets using remap
#-------------------------------------------------------------------------------
promoter.gene.dnam.pair <- MethReg::get_region_target_gene(
  rownames(dnam) %>% MethReg::make_granges_from_names(),
  method = "genes.promoter.overlap",
  genome = "hg19"
)

# 317025 cpgs are promoter
nrow(promoter.gene.dnam.pair) # 317025
promoter.gene.dnam.pair <- promoter.gene.dnam.pair %>% dplyr::filter(.data$target %in% rownames(expression.matrix))

# 175036 cpgs are promoter
nrow(promoter.gene.dnam.pair) # 175036
promoter.gene.dnam.pair$probeID <- names(EPIC.hg19)[match(promoter.gene.dnam.pair$regionID,make_names_from_granges(EPIC.hg19))]
promoter.gene.dnam.pair <- promoter.gene.dnam.pair %>% dplyr::filter(.data$regionID %in% rownames(dnam))
nrow(promoter.gene.dnam.pair) # 175036

results.promoter.analysis <- plyr::adply(
  .data = promoter.gene.dnam.pair,
  .margins = 1,
  .fun = function(row){
    auxfunction(row = row,exp = expression.matrix, dnam = dnam)    
  }
)

results.promoter.analysis <- add_cpgs_from_and_do_fdr(results.promoter.analysis)
```

### Match with the reliable scores

```{r eval = F}
# Match with the reliable scores
results.promoter.analysis$cpg <- results.promoter.analysis$probeID
results.promoter.analysis$probeID <- NULL
results.promoter.analysis.with.reliable <- left_join(
  results.promoter.analysis,
  reliable_df
)
# Group by absolute correlation
results.promoter.analysis.with.reliable <- results.promoter.analysis.with.reliable %>% 
  mutate(rho.group = ifelse(abs(rho) < 0.4, "< 0.4", 
         ifelse(abs(rho) >= 0.4 & abs(rho) < 0.6, "0.4 - 0.6", 
                       ifelse(abs(rho) >= 0.6 & abs(rho) < 0.75, "0.6 - 0.75", "> 0.75"))))
results.promoter.analysis.with.reliable$rho.group <- factor(results.promoter.analysis.with.reliable$rho.group,
                                                            levels = c("< 0.4", "0.4 - 0.6", "0.6 - 0.75", "> 0.75"))
```

```{r eval = F}
#-------------------------------------------------------------------------------
# Save results
#-------------------------------------------------------------------------------
write_csv(
  results.promoter.analysis.with.reliable,
  file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_promoter_region.csv")
)
```

```{r message = F}
results.promoter.analysis.with.reliable <- read_csv(file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_promoter_region.csv"))
```


```{r fig.width=9,fig.height=5, echo = F}
ggboxplot(
  results.promoter.analysis.with.reliable,
  x = "rho.group",
  y = "ICC_Beta",
  ylab = "ICC",
  xlab = "Group of absolute correlation for RNA and DNAm in promoter region",
  fill = "rho.group",
  palette = "jco",
  outlier.shape = NA
)
```

```{r fig.width=9,fig.height=5, echo = F}
results.promoter.analysis.with.reliable <- results.promoter.analysis.with.reliable %>% 
  mutate(sig = ifelse(p.val < 1e-05, "Yes", "No"))
ggboxplot(
  results.promoter.analysis.with.reliable,
  x = "sig",
  y = "ICC_Beta",
  ylab = "ICC",
  xlab = "Significant DNAm-mRNA association in promoter region ",
  fill = "sig",
  palette = "jco",
  outlier.shape = NA
) + guides(fill = "none")
ggsave(filename = file.path(dir.plot, "ADNI_DNAm_RNA_vs_ICC_promoter_region_boxplot.pdf"),
       width = 8, height = 5)
```

## Distal analysis

### Calculate correlations

```{r eval = F}
# -------------------------------------------------------------------------------
# Distal analysis, triplets using remap
# -------------------------------------------------------------------------------
distal.gene.dnam.pair <- MethReg::get_region_target_gene(
  rownames(dnam) %>% MethReg::make_granges_from_names(),
  method = "nearby.genes",
  num.flanking.genes = 10,
  rm.promoter.regions.from.distal.linking = TRUE,
  genome = "hg19"
)

# 7905270 cpgs are distal
nrow(distal.gene.dnam.pair) # 7905270
distal.gene.dnam.pair <- distal.gene.dnam.pair %>% dplyr::filter(.data$target %in% rownames(expression.matrix))

# 2522331 cpgs are distal
nrow(distal.gene.dnam.pair) # 2522331
distal.gene.dnam.pair$probeID <- names(EPIC.hg19)[match(distal.gene.dnam.pair$regionID,make_names_from_granges(EPIC.hg19))]
distal.gene.dnam.pair <- distal.gene.dnam.pair %>% dplyr::filter(.data$regionID %in% rownames(dnam))
nrow(distal.gene.dnam.pair) # 2522331

# To reduce memory, we calculate the association separately and run 100,000 correlations each time
results.distal.analysis <- list()
doParallel::registerDoParallel(4)
k <- seq(1,nrow(distal.gene.dnam.pair), by = 100000)
k <- c(k, nrow(distal.gene.dnam.pair) + 1)
for(i in (4:(length(k) - 1))){
  results.distal.analysis[[i]] <- plyr::adply(
    .data = distal.gene.dnam.pair[k[i] : (k[i + 1] - 1),],
    .margins = 1,
    .fun = function(row){
      print(i)
      return(auxfunction(row = row,exp = expression.matrix, dnam = dnam))
    },.parallel = T
  )
}
results.distal.analysis.df <- results.distal.analysis %>% Reduce(rbind,.)
results.distal.analysis.df <- results.distal.analysis.df %>% add_cpgs_from_and_do_fdr()
```

```{r eval = F}
#-------------------------------------------------------------------------------
# Save results
#-------------------------------------------------------------------------------
write_csv(
  results.distal.analysis.df,
  file.path(dir.RNA_vs_DNAm,"ADNI_cpg_Target_vs_DNAm_Distal_10_up_10_down.csv")
)
```

### Select most significant association gene

```{r eval = F}
# Select the most significant associated genes based on the p-value 
results.distal.analysis.df2 <- results.distal.analysis.df %>% 
  group_by(probeID) %>% slice_min(p.val)
```

### Match with the reliable scores

```{r eval = F}
# Match with the reliable scores
results.distal.analysis.df2$cpg <- results.distal.analysis.df2$probeID
results.distal.analysis.df2$probeID <- NULL
results.distal.analysis.with.reliable <- left_join(
  results.distal.analysis.df2,
  reliable_df
)
# Group by absolute correlation
results.distal.analysis.with.reliable <- results.distal.analysis.with.reliable %>% 
  mutate(rho.group = ifelse(abs(rho) < 0.4, "< 0.4", 
         ifelse(abs(rho) >= 0.4 & abs(rho) < 0.6, "0.4 - 0.6", 
                       ifelse(abs(rho) >= 0.6 & abs(rho) < 0.75, "0.6 - 0.75", "> 0.75"))))
results.distal.analysis.with.reliable$rho.group <- factor(results.distal.analysis.with.reliable$rho.group,
                                                            levels = c("< 0.4", "0.4 - 0.6", "0.6 - 0.75", "> 0.75"))
```

```{r eval = F}
#-------------------------------------------------------------------------------
# Save results
#-------------------------------------------------------------------------------
write_csv(
  results.distal.analysis.with.reliable,
  file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_distal_region.csv")
)
```

```{r message = F}
results.distal.analysis.with.reliable <- read_csv(file.path(dir.RNA_vs_DNAm,"Reliable_ADNI_cpg_Target_vs_DNAm_distal_region.csv"))
```


```{r fig.width=9,fig.height=5, echo = F}
ggboxplot(
  results.distal.analysis.with.reliable,
  x = "rho.group",
  y = "ICC_Beta",
  ylab = "ICC",
  xlab = "Group of absolute correlation for RNA and DNAm in distal region",
  fill = "rho.group",
  palette = "jco",
  outlier.shape = NA
)
```

```{r fig.width=9,fig.height=5, echo = F}
results.distal.analysis.with.reliable <- results.distal.analysis.with.reliable %>% 
  mutate(sig = ifelse(p.val < 1e-05, "Yes", "No"))
ggboxplot(
  results.distal.analysis.with.reliable,
  x = "sig",
  y = "ICC_Beta",
  ylab = "ICC",
  xlab = "Significant DNAm-mRNA association in distal region",
  fill = "sig",
  palette = "jco",
  outlier.shape = NA
) + guides(fill = "none")
ggsave(filename = file.path(dir.plot, "ADNI_DNAm_RNA_vs_ICC_distal_region_boxplot.pdf"),
       width = 8, height = 5)
```


# Session Information

```{r}
devtools::session_info()
```

